src/script.js

```diff
@@ -22,31 +22,74 @@ export function config (filePath) {
 }
 
 export function load (runfilePath, config, logger, requirer, access) {
-  // try to load babel-register
-  try {
-    logger.log('Requiring babel-register...')
-    if (config['babel-register']) {
+  let babelEnabled = false
+  let tsEnabled = false
+  let hasJsRunfile = false
+  let hasTsRunfile = false
+  let importError = null
+
+  if (config['babel-register']) {
+    try {
       requirer(config['babel-register'])
-    } else {
+      babelEnabled = true
+    } catch (error) {
+      importError = error
+    }
+  } else if (config['ts-register']) {
+    try {
+      requirer(config['ts-register'])
+      tsEnabled = true
+    } catch (error) {
+      importError = error
+    }
+  } else {
+    try {
       requirer('./node_modules/babel-register')
+      babelEnabled = true
+    } catch (error) {
+      importError = error
     }
-  } catch (error) {
-    logger.log('Requiring failed. Fallback to pure node.')
-    if (config['babel-register']) {
-      throw error
+
+    if (!babelEnabled) {
```

Comment:
I think that we probably should drop "automagical" transpiler detection. It will get code unnecessary complicated and also new versions of node support more and more es6/es7 features so usage without transpilers starts to be wider.

Having config:

```json
"runjs": {
  "transpiler": "ts-node/register",
  "runfile": "./runfile.ts"
}
```

is really not much and as a side effect it will allow to handle easily other transpilers like for example CoffeeScript, without making any exceptions/condition within runjs code.

We would need to roll with `4.x` version though as that change will break Babel working for existing setups (by not having config in `package.json`)